tests/components/rainforest_raven/test_coordinator.py

```diff
@@ -18,6 +19,7 @@
 async def test_coordinator_device_info(hass: HomeAssistant) -> None:
     """Test reporting device information from the coordinator."""
     entry = create_mock_entry()
+    entry._async_set_state(hass, ConfigEntryState.SETUP_IN_PROGRESS, None)
```

Comment:
Would be nice to refactor these tests to not depend on integration details like the coordinator since these tests pop up failing all the time when we refactor things in helpers.

Reply:
> Would be nice to refactor these tests to not depend on integration details like the coordinator since these tests pop up failing all the time when we refactor things in helpers.

First step: #128517 moves to runtime data

Reply:
> > Would be nice to refactor these tests to not depend on integration details like the coordinator since these tests pop up failing all the time when we refactor things in helpers.
> 
> First step: #128517 moves to runtime data

Second step: #128591 adds abstraction to the initialisation and device_registry tests